core/dbt/clients/yaml_helper.py

```diff
@@ -1,17 +1,16 @@
 import dbt.exceptions
 from typing import Any, Dict, Optional
-import yaml
-import yaml.scanner
+import oyaml as yaml
```

Comment:
I replaced yaml with oyaml in order to retain ordering when prompting for user input. Rather than keep both I just replaced every reference with oyaml. It may well be preferred that we leave all other imports as-is, let me know.

Reply:
@kwigley do you have any thoughts here?

Reply:
It looks like the switch from `yaml` to `oyaml` has broken one highly specific test, which checks the sorting behavior of the `toyaml` Jinja context method. The `sort_keys` argument is not being respected by `oyaml.safe_dump`. Glad we have a test for it!

https://github.com/dbt-labs/dbt/blob/3789acc5a7b3f71b4e333ac6e235c62ee0c957f5/test/integration/013_context_var_tests/tests/to_yaml.sql#L5

https://github.com/dbt-labs/dbt/blob/3789acc5a7b3f71b4e333ac6e235c62ee0c957f5/core/dbt/context/base.py#L416

I think my preference would probably be to avoid switching from `yaml` to `oyaml` wherever possible. I'm also wondering if there's another way we can preserve prompt order for `target_options.yml`, even if it requires an extra attribute